device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -91,7 +91,7 @@ __global__ void find_tracks(
     const unsigned int step, const unsigned int n_max_candidates,
     bound_track_parameters_collection_types::view out_params_view,
     vecmem::data::vector_view<candidate_link> links_view,
-    unsigned int& n_candidates) {
+    unsigned int* n_candidates) {
```

Comment:
To be clear, I'm absolutely happy about making this into a pointer. On the `__global__` function we should use pointers indeed, as it makes the intentions clearer. But on `__device__` functions I much prefer references.